geodiff/src/sqlitedriver.cpp

```diff
@@ -0,0 +1,576 @@
+/*
+ GEODIFF - MIT License
+ Copyright (C) 2020 Martin Dobias
+*/
+
+#include "sqlitedriver.h"
+
+#include "geodiffchangeset.h"
+
+#include <memory.h>
+
+void SqliteDriver::open( const std::map<std::string, std::string> &conn )
+{
+  auto connBaseIt = conn.find( "base" );
+  if ( connBaseIt == conn.end() )
+    throw GeoDiffException( "Missing 'base' file" );
+
+  auto connModifiedIt = conn.find( "modified" );
+  mHasModified = connModifiedIt != conn.end();
+
+  std::string base = connBaseIt->second;
+
+  db = std::make_shared<Sqlite3Db>();
+  if ( mHasModified )
+  {
+    db->open( connModifiedIt->second );
+
+    Buffer sqlBuf;
```

Comment:
I am thinking to create another constructor for bufffer with the string, so instead of constructor and printf you can just do it directly? Or maybe even do override of db->exec function for easier code for caller?